lib/dao.js

```diff
@@ -840,6 +840,11 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId && id) {
```

Comment:
Actually you do, as you can have `forceId` set to true and `id` is null or undefined, and it fails some cases .. Anyways I got to review the test cases first before I make the assumption